tiny_dnn/util/serialization_functions.h

```diff
@@ -545,295 +566,303 @@ struct serialization_buddy {
-       cereal::make_nvp("window_width", params_.weight.width_),
-       cereal::make_nvp("window_height", params_.weight.height_),
-       cereal::make_nvp("out_channels", params_.out.depth_),
-       cereal::make_nvp("connection_table", params_.tbl),
-       cereal::make_nvp("pad_type", params_.pad_type),
-       cereal::make_nvp("has_bias", params_.has_bias),
-       cereal::make_nvp("w_stride", params_.w_stride),
-       cereal::make_nvp("h_stride", params_.h_stride));
+    arc(ar, make_nvp("in_size", params_.in),
+        make_nvp("window_width", params_.weight.width_),
+        make_nvp("window_height", params_.weight.height_),
+        make_nvp("out_channels", params_.out.depth_),
+        make_nvp("connection_table", params_.tbl),
+        make_nvp("pad_type", params_.pad_type),
+        make_nvp("has_bias", params_.has_bias),
+        make_nvp("w_stride", params_.w_stride),
+        make_nvp("h_stride", params_.h_stride));
   }
 
   template <class Archive>
   static inline void serialize(
       Archive &ar, tiny_dnn::quantized_fully_connected_layer &layer) {
     layer.serialize_prolog(ar);
     auto &params_ = layer.params_;
-    ar(cereal::make_nvp("in_size", params_.in_size_),
-       cereal::make_nvp("out_size", params_.out_size_),
-       cereal::make_nvp("has_bias", params_.has_bias_));
+    arc(ar, make_nvp("in_size", params_.in_size_),
+        make_nvp("out_size", params_.out_size_),
+        make_nvp("has_bias", params_.has_bias_));
   }
 
   template <class Archive>
   static inline void serialize(Archive &ar,
                                tiny_dnn::recurrent_cell_layer &layer) {
     layer.serialize_prolog(ar);
     auto &params_ = layer.params_;
-    ar(cereal::make_nvp("in_size", params_.in_size_),
-       cereal::make_nvp("out_size", params_.out_size_),
-       cereal::make_nvp("has_bias", params_.has_bias_),
-       cereal::make_nvp("activation", params_.activation_));
+		arc(ar,
+				make_nvp("in_size", params_.in_size_),
+        make_nvp("out_size", params_.out_size_),
+        make_nvp("has_bias", params_.has_bias_),
+				make_nvp("activation", params_.activation_));
   }
 
   template <class Archive>
   static inline void serialize(Archive &ar, tiny_dnn::slice_layer &layer) {
     layer.serialize_prolog(ar);
-    ar(cereal::make_nvp("in_size", layer.in_shape_),
-       cereal::make_nvp("slice_type", layer.slice_type_),
-       cereal::make_nvp("num_outputs", layer.num_outputs_));
+    arc(ar,
+				make_nvp("in_size", layer.in_shape_),
+        make_nvp("slice_type", layer.slice_type_),
+        make_nvp("num_outputs", layer.num_outputs_));
   }
 
   template <class Archive>
   static inline void serialize(Archive &ar, tiny_dnn::sigmoid_layer &layer) {
     layer.serialize_prolog(ar);
-    ar(cereal::virtual_base_class<tiny_dnn::activation_layer>(&layer),
-       cereal::make_nvp("in_size", layer.in_shape()[0]));
+    arc(ar, 
+				cereal::virtual_base_class<tiny_dnn::activation_layer>(&layer),
```

Comment:
/cc @beru @nyanp @prlz77 something is wrong with `virtual_base_class` after the rebase